GoogleSheetI18n.Api.SimpleWebApi/ClientApp/src/components/Administration.js

```diff
@@ -5,51 +5,75 @@ import { useTranslation } from 'react-i18next';
-      (error) => {
-        setIsLoaded(true);
-        setError(error);
-      }
-    );
-    setIsLoaded(false);
-  }, [authState, setIsLoaded, setError]);
+    const clearCache = useCallback(() => {
+        i18nService.reloadLocalStore(
+            authState.token
+        ).then(
+            (result) => {
+                if (result.status !== 200) {
+                    setIsLoaded(true);
+                    alert(`Error when clearing cache. HTTP status: ${result.status}`);
+                    return;
+                }
+                window.location.reload();
+                setIsLoaded(true);
+            },
+            (error) => {
+                setIsLoaded(true);
+                setError(error);
+            }
+        );
+        setIsLoaded(false);
+    }, [authState, setIsLoaded, setError]);
+
+    const validateSpreadsheet = useCallback(type => {
+        i18nService.validateSpreadsheet(type)
+            .then(res => res.json())
+            .then(
+                (result) => {
+                    setIsLoaded(true);
+                    if (result.message === "") {
+                        alert(`Validation was successful`);
+                        return;
+                    }   
+                    setError(result);
+                },
+                (error) => {
+                    setIsLoaded(true);
+                    setError(error);
+                }
+        );
+        setIsLoaded(false);
+    }, [setIsLoaded, setError]);
 
-  if (error) {
-    return <ErrorSummary message={error.message} />;
-  }
+    if (error) {
+        return <ErrorSummary message={error.message} />;
+    }
 
-  return (
-    <div>
-      <h1>{t('admin-page:administration')}</h1>
-      <h3>{t('admin-page:basic-actions')}</h3>
-      {isLoaded && <div>
-        <a href={linkToGoogleSheet} target="_blank" rel="noreferrer" className="btn btn-lg btn-success mb-3 mr-2">{t('admin-page:view-google-sheet')}</a>
-        {authState.isAuthenticated && <button onClick={clearCache} className="btn btn-lg btn-outline-secondary mb-3">
-          {t('admin-page:reload-local-store')}
-        </button>}
-      </div>}
-      {!isLoaded && <Loader />}
-    </div>
-  );
+    return (
+        <div>
+            <h1>{t('admin-page:administration')}</h1>
+            <h3>{t('admin-page:basic-actions')}</h3>
+            {isLoaded && <div>
+                <a href={linkToGoogleSheet} target="_blank" rel="noreferrer" className="btn btn-lg btn-success mb-3 mr-2">{t('admin-page:view-google-sheet')}</a>
+                {authState.isAuthenticated && <button onClick={clearCache} className="btn btn-lg btn-outline-secondary mb-3">
+                    {t('admin-page:reload-local-store')} </button>}
+                {authState.isAuthenticated && <button onClick={() => validateSpreadsheet(spreadsheetTypes.LOCAL)} className="btn btn-lg btn-outline-secondary mb-3 btn-validation">
+                    {t('admin-page:validation-local-store')} </button>}
```

Comment:
Could you change button name from `validation` to `validate`, in other words using verbs not nouns for button names.

Reply:
renamed